sympy/controlsys/lti.py

```diff
@@ -0,0 +1,757 @@
+
+        Returns the representation in state space of a given transfer function
+
+        Parameters
+        ==========
+
+        G: Matrix
+            Matrix valued transfer function G(s) in laplace space
+        s: symbol
+            variable s, where G is dependent from
+
+        See Also
+        ========
+
+        Utils : some quick tools for matrix polynomials
+
+        References
+        ==========
+
+        Joao P. Hespanha, Linear Systems Theory. 2009.
+        """
+
+        A, B, C, D = 4 * [None]
+
+        try:
+            m, k = G.shape
+
+        except AttributeError:
+            raise TypeError("G must be a matrix")
+
+        # test if G is proper
+        if not _is_proper(G, s, strict=False):
+            raise ValueError("G must be proper!")
+
+        # define D as the limit of G for s to infinity
+        D = G.limit(s, oo)
+
+        # define G_sp as the (stricly proper) difference of G and D
+        G_sp = simplify(G - D)
+
+        # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+        # get the coefficients of the monic least common denominator of all entries of G_sp
+        # compute a least common denominator using utl and lcm
+        lcd = lcm(_fraction_list(G_sp, only_denoms=True))
+
+        # make it monic
+        lcd = simplify(lcd / LC(lcd, s))
+
+        # and get a coefficient list of its monic. The [1:] cuts the LC away (thats a one)
+        lcd_coeff = Poly(lcd, s).all_coeffs()[1:]
+
+        # get the degree of the lcd
+        lcd_deg = degree(lcd, s)
+
+        # get the Matrix Valued Coeffs of G_sp in G_sp = 1/lcd * (N_1 * s**(n-1) + N_2 * s**(n-2) .. +N_n)
+        G_sp_coeff = _matrix_coeff(simplify(G_sp * lcd), s)
+        G_sp_coeff = [zeros(m, k)] * (lcd_deg - len(G_sp_coeff)) + G_sp_coeff
+
+        # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+        # now store A, B, C, D in terms of the coefficients of lcd and G_sp
+        # define A
+        A = (-1) * lcd_coeff[0] * eye(k)
+
+        for alpha in lcd_coeff[1:]:
+            A = A.row_join((-1) * alpha * eye(k))
+
+        for i in xrange(lcd_deg - 1):
+            if i == 0:
+                tmp = eye(k)
+            else:
+                tmp = zeros(k)
+
+            for j in range(lcd_deg)[1:]:
+                if j == i:
+                    tmp = tmp.row_join(eye(k))
+                else:
+                    tmp = tmp.row_join(zeros(k))
+            if tmp is not None:
+                A = A.col_join(tmp)
+
+        # define B
+        B = eye(k)
+        for i in xrange(lcd_deg - 1):
+            B = B.col_join(zeros(k))
+
+        # define C
+        C = G_sp_coeff[0]
+        for i in range(lcd_deg)[1:]:
+            C = C.row_join(G_sp_coeff[i])
+
+        # return the state space representation
+        return [simplify(A), simplify(B), simplify(C), simplify(D)]
```

Comment:
All of the internal simplify functions in this command should be removed. The user can call simplify on the result if they choose.